pkg/probe/probe_registration.go

```diff
@@ -62,49 +57,108 @@ type DiscoveryMetadata struct {
 // Performance discovery is set to -1 implies that the performance discovery is not supported by the probe.
 func NewDiscoveryMetadata() *DiscoveryMetadata {
 	return &DiscoveryMetadata{
-		fullDiscovery:        DEFAULT_FULL_DISCOVERY_IN_SECS,
-		incrementalDiscovery: DISCOVERY_NOT_SUPPORTED,
-		performanceDiscovery: DISCOVERY_NOT_SUPPORTED,
+		fullDiscovery:        pkg.DEFAULT_FULL_DISCOVERY_IN_SECS,
+		incrementalDiscovery: pkg.DISCOVERY_NOT_SUPPORTED,
+		performanceDiscovery: pkg.DISCOVERY_NOT_SUPPORTED,
+	}
+}
+
+type DiscoveryMetadataOption func(*DiscoveryMetadata)
+
+func CreateDiscoveryMetadata(options ...DiscoveryMetadataOption) *DiscoveryMetadata {
```

Comment:
I feel that this is unnecessary more complicated. Why not use the `DiscoveryMetadata.setXXX()` directly?